server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -899,7 +899,7 @@ public void commitSegmentInfos(SegmentInfos latestSegmentInfos, long maxSeqNo, l
             final Map<String, String> userData = new HashMap<>(latestSegmentInfos.getUserData());
             userData.put(LOCAL_CHECKPOINT_KEY, String.valueOf(processedCheckpoint));
             userData.put(SequenceNumbers.MAX_SEQ_NO, Long.toString(maxSeqNo));
-            latestSegmentInfos.setUserData(userData, true);
+            latestSegmentInfos.setUserData(userData, false);
```

Comment:
Can you add a unit test to verify post change SegmentInfos verison is not bumped.

Reply:
Sure, I'll add another test in StoreTests, the test I added `testSegmentInfosAndReplicationCheckpointTuple` will also fail without this change when comparing primary/replica checkpoints after a flush.